test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs

```diff
@@ -16,51 +18,123 @@ public HttpClientTests(ITestOutputHelper output)
             {
                 Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode}");
 
                 var spans = agent.WaitForSpans(1);
                 Assert.True(spans.Count > 0, "expected at least one span");
 
+                var traceId = GetHeader(processResult.StandardOutput, HttpHeaderNames.TraceId);
+                var parentSpanId = GetHeader(processResult.StandardOutput, HttpHeaderNames.ParentId);
+
                 // inspect the top-level span, underlying spans can be HttpMessageHandler in .NET Core
-                foreach (var span in spans.Take(1))
-                {
-                    Assert.Equal("http.request", span.Name);
-                    Assert.Equal("Samples.HttpMessageHandler", span.Service);
-                    Assert.Equal(SpanTypes.Http, span.Type);
-                    Assert.Equal(nameof(WebRequest), span.Tags[Tags.InstrumentationName]);
-                }
+                var firstSpan = spans.First();
+                Assert.Equal("http.request", firstSpan.Name);
+                Assert.Equal("Samples.HttpMessageHandler-http-client", firstSpan.Service);
+                Assert.Equal(SpanTypes.Http, firstSpan.Type);
+                Assert.Equal(nameof(WebRequest), firstSpan.Tags[Tags.InstrumentationName]);
+
+                var lastSpan = spans.Last();
+                Assert.Equal(lastSpan.TraceId.ToString(CultureInfo.InvariantCulture), traceId);
+                Assert.Equal(lastSpan.SpanId.ToString(CultureInfo.InvariantCulture), parentSpanId);
             }
         }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        public void WebClient_TracingDisabled()
+        {
+            int agentPort = TcpPortProvider.GetOpenPort();
+            int httpPort = TcpPortProvider.GetOpenPort();
+
+            using (var agent = new MockTracerAgent(agentPort))
+            using (ProcessResult processResult = RunSampleAndWaitForExit(agentPort, $"WebClient TracingDisabled Port={httpPort}"))
+            {
+                Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode}");
+
+                var spans = agent.WaitForSpans(1, 500);
+                Assert.Equal(0, spans.Count);
+
+                var traceId = GetHeader(processResult.StandardOutput, HttpHeaderNames.TraceId);
+                var parentSpanId = GetHeader(processResult.StandardOutput, HttpHeaderNames.ParentId);
+                var tracingEnabled = GetHeader(processResult.StandardOutput, HttpHeaderNames.TracingEnabled);
+
+                Assert.Null(traceId);
+                Assert.Null(parentSpanId);
+                Assert.Equal("false", tracingEnabled);
+            }
+        }
+
+        private string GetHeader(string stdout, string name)
+        {
+            var pattern = $@"^\[HttpListener\] request header: {name}=(\w+)\r?$";
```

Comment:
I'm not extremely fond of the raw text + regexp protocol but it works 😄 If we start doing it for other integration tests I'd be in favor of defining a more structured format though.

Reply:
I agree! This was faster since I was already sending this text to stdout. We should revisit and come up with a better system. I created a Trello card to follow up.